VideoLocker/src/main/java/org/edx/mobile/event/NewVersionAvailableEvent.java

```diff
@@ -0,0 +1,193 @@
+     *
+     * @return Whether the current version is unsupported.
+     */
+    public boolean isUnsupported() {
+        return isUnsupported;
+    }
+
+    /**
+     * Resolve the notification string, and return it.
+     *
+     * @param context A Context to resolve the string
+     * @return The notification string.
+     */
+    @NonNull
+    public CharSequence getNotificationString(@NonNull final Context context) {
+        @StringRes
+        final int notificationStringRes;
+        if (isUnsupported) {
+            notificationStringRes = R.string.app_version_unsupported;
+        } else if (lastSupportedDate == null) {
+            notificationStringRes = R.string.app_version_outdated;
+        } else {
+            // Deadline date is available, but won't be displayed for now.
+            notificationStringRes = R.string.app_version_deprecated;
+        }
+        return context.getText(notificationStringRes);
+    }
+
+    /**
+     * Mark the event as consumed by the subscribers.
+     */
+    public void markAsConsumed() {
+        isConsumed = true;
+    }
+
+    /**
+     * @return Whether the event has been consumed by the subscribers.
+     */
+    public boolean isConsumed() {
+        return isConsumed;
+    }
+
+    /**
+     * Compare this to another instance to determine their priority. Events reporting the current
+     * app version as unsupported have the highest priority, followed by deprecation events, which
+     * are prioritised according to the closeness of the last supported date they report, followed
+     * by new version availability events, which are prioritized according to the reported new
+     * version number.
+     *
+     * @param another the object to compare to this instance.
+     * @return a negative integer if this instance has lesser priority than {@code another};
+     *         a positive integer if this instance has greater priority than {@code another};
+     *         0 if this instance has the same priority as {@code another}.
+     * @throws ClassCastException if {@code another} cannot be converted into
+     *                            something comparable to {@code this} instance.
+     */
+    @Override
+    public int compareTo(@NonNull final NewVersionAvailableEvent another) {
+        int result = BooleanUtils.compare(isUnsupported, another.isUnsupported);
+        if (result == 0) {
+            /* Reverse the comparator here, since the closer the date is, the higher the priority.
+             * Since we're reversing the result, the comparator is instructed to count a null value
+             * in the last supported date (non-deprecated new version availability event) as of
+             * higher priority than a non-null value (deprecation event).
+             */
+            result = 0 - ObjectUtils.compare(lastSupportedDate, another.lastSupportedDate, true);
+            if (result == 0) {
+                result = ObjectUtils.compare(newVersion, another.newVersion);
```

Comment:
> I think this would be more legible if we had the null inline anyway.

Done in commit 4a4bd30be8b792670912d812c9b466b1aa7781bf.

> It's not worth importing these classes to avoid a couple null checks here.

Imports reverted in commit efb2bf4b3469d4eab136be2c94df36c56ad1d70d, and new ticket created for fixing the existing references to the compile-only Commons Lang 2.4 dependency at [MA-2682][].

[MA-2682]: https://openedx.atlassian.net/browse/MA-2682